accepted/00102-upgrade-to-salt-3008.md

```diff
@@ -0,0 +1,199 @@
+
+Particularly, this new 3008.x LTS release is the first version after [the great module migration](https://salt.tips/the-great-salt-module-migration/), and lot of Salt modules have been purged from the main Salt package in favor of [Salt Extensions](https://github.com/salt-extensions), including for example, the `zypperpkg` and `transactional_update` modules.
+
+At this point in time, only few modules have been already migrated to Salt Extensions, a lot of other modules are still waiting at [community-extensions-holding repository](https://github.com/salt-extensions/community-extensions-holding) to be migrated.
+
+The purpose of this RFC is to define:
+- Salt and Salt Extensions packaging strategy.
+- How to customize the Salt package with more extensions.
+- New dependencies required for Salt.
+- How to release Salt 3008.x to SLE.
+- How to maintain Salt 3008.x going forward.
+
+# Detailed design
+[design]: #detailed-design
+
+## Packaging
+
+Due the great module purge, we need to rethink the way we package Salt, as lot of the modules are been dropped and Salt Extensions are stepping in now.
+
+As upstream Salt project wants to drops the support from their side in favor of [community driven support](https://github.com/salt-extensions#what-are-salt-extensions) and lifecycle for the different Salt Extensions, this raises the question about which of them do we want to include and support.
+
+At the time of writting this RFC, most of the modules dropped from the Salt core are not yet migrated to Salt Extensions, therefore nobody from the community has claimed their ownership yet.
+
+### Builtin Salt Extensions
+
+There are a couple of Salt Extensions that we want to have integrated by default in our main `python311-salt` package. These are:
+- zypperpkg
+- dpkg
+- transactional_update
+- openscap
+- docker / dockerng
+- ...
+
+These are the minimum required extensions to be able to do basic operations in SUSE/openSUSE distributions and also allow basic operations on the context of SUSE Manager clients. We must maintain and support these builtin Salt Extensions.
+
+This list could be reduced even more by just providing the `zypperpkg` and `transactional_update` modules and then providing the rest via `/usr/share/susemanager/salt/[_modules,_states,...]` in the Uyuni / SUSE Manager server.
+
+Builtin Salt Extensions will be integrated into our main "openSUSE/salt" GitHub repo codebase for 3008.x, and provided as content of the main `python311-salt` package.
+
+#### An example of the Git repo:
+```
+salt
+|-- builin-extensions
+|   |-- saltext-zypperpkg
+|   |-- saltext-dpkg
+|   |-- ...
+...
+```
+
+The idea would be to include the releases for the different builtin extension directly inside our GH repo, as it was previously done with "tornado". We should take this oportunity also to take the ownership, migrate and create the official `saltext-zypperpkg` extension and probably some others.
```

Comment:
How do we do scm on these directories? How do make sure we contribute our changes upstream and then pull them down? How do we publish extensions maintained by us to PyPI?

Reply:
Yeah, all these questions must be still clarified and defined in a new maintaining worflow for Salt.

I added a note on the current RFC mentioning that we are covering the "new maintining workflow for Salt" in a separated RFC, so we can agree on the proposed plan here for the Salt 3008 upgrade, and start pushing it forward while we keep discussing and designing the new maintaining workflow.